### PR TITLE
TypeScript: improve handling of computed properties

### PR DESCRIPTION
**Pull request by azz**
#1480

### CODE REVIEW

File: src/printer.js [hunk 1]
```diff
@@ -2080,8 +2080,20 @@ function genericPrintNoParens(path, options, print, args) {
     case "TSArrayType":
       return concat([path.call(print, "elementType"), "[]"]);
     case "TSPropertySignature":
+      var computed = !namedTypes.Identifier.check(n.name);
```
Comment (azz) [inline src/printer.js]: Note that this change caused the test
`tests/typescript/conformance/types/union/__snapshots__/jsfmt.spec.js.snap` to add square brackets around the numbers (which is correct, but not required).

Comment (vjeux) [inline src/printer.js]: We can figure out how to make it print well once it's correct :)

**Comment by vjeux** [comment]
Arg there are conflicts :(

**Comment by azz** [comment]
Rebased. Might want to wait for Travis because I didn't re-run the tests locally.

**Pull request merged**